### PR TITLE
feat(testing): adds goleak check to each suite and template

### PR DESCRIPTION
**Pull request by bartoszmajsak**
#### Short description of what this resolves:

I've added goleak when testing watch functionality - helped me finding a few corner cases in this part.

We somehow forgot to add that across the whole test suite.

#### Changes proposed in this pull request:

- adds `AfterSuite` in each `_suite_test.go` with calls to goleak detection
- enhances ginkgo template so that new suites will have that check right from the beginning

### CODE REVIEW

File: cmd/ike/session/session_suite_test.go [hunk 1]
```diff
@@ -13,3 +15,10 @@ func TestSession(t *testing.T) {
 	RegisterFailHandler(Fail)
 	RunSpecWithJUnitReporter(t, "Session Suite")
 }
+
+var _ = SynchronizedAfterSuite(func() {}, func() {
+	goleak.VerifyNone(GinkgoT(),
```
Comment (aslakknutsen) [inline cmd/ike/session/session_suite_test.go]: At this point, one can start to wonder about that one 'global' suite..

Comment (bartoszmajsak) [reply cmd/ike/session/session_suite_test.go]: The only problem is that you can only have one "Before/AfterSuite" hook per suite, so we would need to put all the logic there, does not matter if it's used by every test or not. A bit of a disadvantage.
